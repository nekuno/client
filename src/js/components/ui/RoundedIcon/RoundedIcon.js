import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './RoundedIcon.scss';

export default class RoundedIcon extends Component {

    static propTypes = {
        icon           : PropTypes.string.isRequired,
        size           : PropTypes.oneOf(['small', 'medium', 'large']).isRequired,
        disabled       : PropTypes.bool,
        background     : PropTypes.string,
        color          : PropTypes.string,
        fontSize       : PropTypes.string,
        onClickHandler : PropTypes.func
    };

    handleClick() {
        if (!this.props.disabled && this.props.onClickHandler) {
            this.props.onClickHandler();
        }
    }

    render() {
        const {icon, size, disabled, background, color, fontSize} = this.props;
        let className = styles.roundedIcon + ' ' + styles[size];
        className = disabled ? styles.disabled + ' ' + className : className;

        return (
            <div className={className} style={{background: background, color: color}} onClick={this.handleClick.bind(this)}>
                <div className={styles.icon + ' icon icon-' + icon} style={{fontSize: fontSize}}/>
            </div>
        );
    }
}