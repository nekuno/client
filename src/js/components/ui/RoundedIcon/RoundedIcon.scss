.rounded-icon {
  background: #6C6F82;
  border: none;
  border-radius: 50%;
  -moz-border-radius: 50%;
  -webkit-border-radius: 50%;
  color: #fff;
  display: inline-block;
  margin: 5px;
  position: relative;
  overflow: hidden;
  cursor: pointer;
  text-align: center;

  &.disabled {
    cursor: default;
    opacity: 0.5;
  }
  &.large {
    height: 75px;
    width: 75px;

    .icon {
      font-size: 75px;
      line-height: 75px;
    }
  }
  &.medium {
    height: 50px;
    width: 50px;

    .icon {
      font-size: 50px;
      line-height: 50px;
    }
  }
  &.small {
    height: 25px;
    width: 25px;

    .icon {
      font-size: 25px;
      line-height: 25px;
    }
  }
}