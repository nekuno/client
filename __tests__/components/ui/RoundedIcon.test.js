import React from 'react';
import RoundedIcon from '../../../src/js/components/ui/RoundedIcon/RoundedIcon.js';
import styles from '../../../src/js/components/ui/RoundedIcon/RoundedIcon.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test RoundedIcon component', () => {
    const mockFn = jest.fn();
    const roundedIcon = shallow(<RoundedIcon icon='facebook' size='medium' onClickHandler={mockFn}/>);
    const disabledRoundedIcon = shallow(<RoundedIcon disabled={true} icon='facebook' size='medium' onClickHandler={mockFn}/>);

    it('should be defined', () => {
        expect(RoundedIcon).toBeDefined();
    });
    it('should render correctly', () => {
        expect(roundedIcon).toMatchSnapshot();
        expect(disabledRoundedIcon).toMatchSnapshot();
    });
    it('RoundedIcon click event', () => {
        roundedIcon.find(`.${styles.roundedIcon}`).simulate('click');
        expect(mockFn).toHaveBeenCalledTimes(1);
    });
    it('RoundedIcon disabled click event', () => {
        disabledRoundedIcon.find(`.${styles.roundedIcon}`).simulate('click');
        expect(mockFn).toHaveBeenCalledTimes(0);
    });
});
