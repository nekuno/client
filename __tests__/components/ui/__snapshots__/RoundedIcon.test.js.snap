// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RoundedIcon component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RoundedIcon
    icon="facebook"
    onClickHandler={[MockFunction]}
    size="medium"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="icon icon icon-facebook"
        style={
          Object {
            "fontSize": undefined,
          }
        }
      />,
      "className": "roundedIcon medium",
      "onClick": [Function],
      "style": Object {
        "background": undefined,
        "color": undefined,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "icon icon icon-facebook",
        "style": Object {
          "fontSize": undefined,
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="icon icon icon-facebook"
          style={
            Object {
              "fontSize": undefined,
            }
          }
        />,
        "className": "roundedIcon medium",
        "onClick": [Function],
        "style": Object {
          "background": undefined,
          "color": undefined,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "icon icon icon-facebook",
          "style": Object {
            "fontSize": undefined,
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Test RoundedIcon component should render correctly 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RoundedIcon
    disabled={true}
    icon="facebook"
    onClickHandler={[MockFunction]}
    size="medium"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="icon icon icon-facebook"
        style={
          Object {
            "fontSize": undefined,
          }
        }
      />,
      "className": "disabled roundedIcon medium",
      "onClick": [Function],
      "style": Object {
        "background": undefined,
        "color": undefined,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "icon icon icon-facebook",
        "style": Object {
          "fontSize": undefined,
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="icon icon icon-facebook"
          style={
            Object {
              "fontSize": undefined,
            }
          }
        />,
        "className": "disabled roundedIcon medium",
        "onClick": [Function],
        "style": Object {
          "background": undefined,
          "color": undefined,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "icon icon icon-facebook",
          "style": Object {
            "fontSize": undefined,
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
